Sources/ComposableArchitecture/Store.swift

```diff
@@ -145,9 +145,10 @@ public final class Store<State, Action> {
   ///     by the reducer.
   public convenience init<R: ReducerProtocol>(
     initialState: @autoclosure () -> R.State,
-    reducer: R,
+    @ReducerBuilder<State, Action> reducer: () -> R,
     prepareDependencies: ((inout DependencyValues) -> Void)? = nil
   ) where R.State == State, R.Action == Action {
+    let reducer = reducer()
```

Comment:
One question is if this should be evaluated using `prepareDependencies`...and if so, does that mean we can drop the `transformDependencies` we chain onto it?

Reply:
There is also the question to forward environmental dependencies around `Store` into the reducer that was raised by @lukeredpath a few weeks ago on Slack. I don't remember exactly what was the outcome. It think this is what was behind my `isEmpty` PR in `swift-dependencies` (the property was used to detect if the environmental dependencies were empty/overridden in order to avoid a trivial `.transformDependencies`).

Reply:
[The discussion in question](https://pointfreecommunity.slack.com/archives/C04L2D0MNJH/p1676982334184199)